src/api/GameContext.tsx

```diff
@@ -26,24 +39,68 @@ type DeleteGameAction = {
   type: 'DELETE_GAME'
 }
 
-type Action = SetCardsAction | DeleteGameAction
+type FinishGameAction = {
+  type: 'FINISH_GAME'
+}
+
+type UpdateButtonAction = {
+  type: 'UPDATE_BUTTON'
+}
+
+type Action =
+  | StartGameAction
+  | SetCardIndexAction
+  | SetCardsAction
+  | DeleteGameAction
+  | FinishGameAction
+  | UpdateButtonAction
 
 type Props = {
   state: State
   dispatch: React.Dispatch<Action>
+  handleButtonClick: () => void
 }
 
 const initialState: State = {
   cards: [],
-  isGameOngoing: false,
+  gameState: GameState.NOT_STARTED,
+  buttonLabel: 'New Game',
+  currentCardIndex: 0,
 }
 
 const gameReducer = (state: State, action: Action): State => {
   switch (action.type) {
+    case 'START_GAME':
+      return { ...state, gameState: GameState.START, currentCardIndex: 0 }
+    case 'SET_CARD_INDEX':
+      return { ...state, currentCardIndex: action.payload }
     case 'SET_CARDS':
-      return { ...state, cards: action.payload, isGameOngoing: true }
+      return {
+        ...state,
+        cards: action.payload,
+        gameState: GameState.ONGOING,
+        currentCardIndex: 0,
+      }
     case 'DELETE_GAME':
-      return { ...state, cards: [], isGameOngoing: false }
+      return {
+        ...state,
+        cards: [],
+        gameState: GameState.NOT_STARTED,
+        currentCardIndex: 0,
+      }
+    case 'FINISH_GAME':
+      return { ...state, gameState: GameState.FINISHED }
+    case 'UPDATE_BUTTON': {
+      let label
+      if (state.gameState === GameState.ONGOING && state.cards.length > 0) {
+        label = 'Solve #' + (state.currentCardIndex + 1)
+      } else if (state.gameState === GameState.FINISHED) {
+        label = 'Finished'
+      } else {
+        label = 'New Game'
+      }
+      return { ...state, buttonLabel: label }
```

Comment:
Könnte man auch gleichzeitig mit dem currentCardIndex aktualisieren um schlankere Kontext-Logik zu haben.

Reply:
done in #41